File: problem/Divide_Array_I/minimumCost.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const minimumCost = A => {
    let α = 51, β = 51;

    for (let i = 1; i < A.length; i++) {
        β = Math.min(β, A[i]);
        if (β < α) [α, β] = [β, α];
        if (α === 1 && β === 1) return A[0] + 2;
    }

    return A[0] + α + β;
};

File: problem/Divide_Array_I/minimumCost.c
int minimumCost(int *A, int ASize)
{
    int a = 51, b = 51;

    for (int i = 1; i < ASize; i++)
    {
        if (A[i] < a)
        {
            b = a;
            a = A[i];
        }
        else if (A[i] < b)
        {
            b = A[i];
        }

        if (a == 1 && b == 1)
            break;
    }

    return A[0] + a + b;
}